### ISSUE TITLE
Compatible with norm colorbar ticks in matplotlib3.5+

### ISSUE THREAD

**Issue by zxdawn** [lukelbd/proplot#322]
### Description

The ticks of colorbar for manual levels are wrong for matplotlib3.5+.

### Steps to reproduce

```python
import proplot as pplt
import matplotlib
import numpy as np

crf_bounds = np.array([0, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 0.97, 0.98, 0.99, 1.0])
norm = matplotlib.colors.BoundaryNorm(boundaries=crf_bounds, ncolors=256)

x = y = np.array([-10, -5, 0, 5, 10])
np.random.seed(20)
data = np.random.uniform(0,1,(4,4))

fig, axs = pplt.subplots()
m = axs.pcolormesh(x, y, data, cmap='RdYlGn', norm=norm)
axs.colorbar([m])
```

**Expected behavior**: [What you expected to happen]

![image](https://user-images.githubusercontent.com/30388627/149529123-a1096a85-d6f3-4a49-a4c8-2147e9204292.png)

**Actual behavior**: [What actually happened]

![image](https://user-images.githubusercontent.com/30388627/149529298-b0fd6074-c408-4091-8671-3c39d09d189f.png)

### Equivalent steps in matplotlib

```python
import matplotlib
import numpy as np
import matplotlib.pyplot as plt

crf_bounds = np.array([0, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 0.97, 0.98, 0.99, 1.0])
norm = matplotlib.colors.BoundaryNorm(boundaries=crf_bounds, ncolors=256)

x = y = np.array([-10, -5, 0, 5, 10])
np.random.seed(20)
data = np.random.uniform(0,1,(4,4))

plt.pcolormesh(x, y, data, cmap='RdYlGn', norm=norm)
plt.colorbar()
```

![image](https://user-images.githubusercontent.com/30388627/149529492-6e7cf034-0af9-4af4-b5b6-3c9a3e099bbe.png)

### Proplot version

Paste the results of `import matplotlib; print(matplotlib.__version__); import proplot; print(proplot.version)`here.

```
3.5.1
0.9.5.post105
```

**Comment by lukelbd** [comment]
Unfortunately this is a matplotlib-specific bug related to minor colorbar ticks in the presence of `BoundaryNorm`. Matplotlib significantly changed the colorblar ticking internals in version 3.5 and it caused some problems. Thought this issue came up in a previous thread... but can't seem to find it.

Here is your example with `cb.minorticks_on()` (feel free to copy this and start a matplotlib issue if you have time):

```python
import matplotlib
import numpy as np
import matplotlib.pyplot as plt

crf_bounds = np.array([0, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 0.97, 0.98, 0.99, 1.0])
norm = matplotlib.colors.BoundaryNorm(boundaries=crf_bounds, ncolors=256)

x = y = np.array([-10, -5, 0, 5, 10])
np.random.seed(20)
data = np.random.uniform(0,1,(4,4))

fig = plt.figure()
ax = fig.add_subplot()
m = ax.pcolormesh(x, y, data, cmap='RdYlGn', norm=norm)
cb = fig.colorbar(m)
cb.minorticks_on()
```

![tmp](https://user-images.githubusercontent.com/19657652/149578420-e22c3075-880f-48e7-bf61-16d0c2d21201.png)

**Comment by zxdawn** [comment]
Thanks, Luke. Created [the issue](https://github.com/matplotlib/matplotlib/issues/22233).

**Comment by lukelbd** [comment]
Lol... had just decided to not be lazy and submit an issue myself (matplotlib/matplotlib#22232). Will delete it.

**Comment by zxdawn** [comment]
Hah, same time for the issue ;)

**Comment by lukelbd** [comment]
Hmm..... on closer inspection I think this is both a proplot "bug" and matplotlib "bug".

It's true that matplotlib's default minor ticking in presence of `BoundaryNorm` is strange, but proplot should be overriding that default behavior by using a `DiscreteLocator` for minor ticks anyway. In your example, you are instead seeing the default `AutoMinorLocator` ticks generated with `cb.minorticks_on()`. I'll reopen and fix this issue.

**Comment by lukelbd** [comment]
Ok, this turned out to be a simple one-line fix (e609f010). 

```python
import proplot as pplt
import matplotlib
import numpy as np
crf_bounds = np.array([0, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 0.97, 0.98, 0.99, 1.0])
norm = matplotlib.colors.BoundaryNorm(boundaries=crf_bounds, ncolors=256)
x = y = np.array([-10, -5, 0, 5, 10])
np.random.seed(20)
data = np.random.uniform(0,1,(4,4))
fig, axs = pplt.subplots()
m = axs.pcolormesh(x, y, data, cmap='RdYlGn', levels=crf_bounds)
cb = axs.colorbar([m])
```
![tmp](https://user-images.githubusercontent.com/19657652/149606016-6d21d4ba-8799-42c3-9f3c-b71c16d8805e.png)
